### DISCUSSION TITLE
Twitch clips. HL service.. What is sha256

### DISCUSSION THREAD

**Discussion by lebedynskyi** [streamlink/streamlink#5696] [Q&A] [answered]
### Description

I am working with investigation abou Twitch clips downloading..
I don't want to use project given app_id for my own purpose..
I found endpoint and data i have to send. 

Here is some input:
Clip URL:
https://clips.twitch.tv/ShyPleasantGullBudStar-0Scj5gLbATa3oVZe

Payload :
{"extensions": {"persistedQuery": {"sha256Hash": "36b89d2507fce29e5ca551df756d27c1cfe079e2609642b4390aa4c35796eb11", "version": 1}}, "operationName": "VideoAccessToken_Clip", "variables": {"slug": "ShyPleasantGullBudStar-0Scj5gLbATa3oVZe"}}

Resulting curl:
curl -X POST https://gql.twitch.tv/gql -H "Client-ID: kimne78kx3ncx6brgo4mv6wki5h1ko" -d '{"extensions": {"persistedQuery": {"sha256Hash": "36b89d2507fce29e5ca551df756d27c1cfe079e2609642b4390aa4c35796eb11", "version": 1}}, "operationName": "VideoAccessToken_Clip", "variables": {"slug": "ShyPleasantGullBudStar-0Scj5gLbATa3oVZe"}}'

I just want to clarify why twitch does not accepty my own application id and what is sha256Hash in payload ?

**Comment by bastimeyer** [comment, marked as answer]
> I just want to clarify why twitch does not accepty my own application id and what is sha256Hash in payload ?

Why are you asking about the Twitch API here? What does Streamlink have to do with your own attempts of trying to access the Twitch GQL API?

`gql.twitch.tv` is **not** a public API. Only their own API client-ID from their website works here, which is `kimne78kx3ncx6brgo4mv6wki5h1ko` (and probably different ones embedded in official apps too). Your own client-IDs are meant for `helix.twitch.tv`, [their public REST API](https://dev.twitch.tv/docs/api/).

`36b89d2507fce29e5ca551df756d27c1cfe079e2609642b4390aa4c35796eb11` is a **persisted GraphQL query ID**, which is pre-compiled on their end for optimization reasons and it's referenced using this hash. The query expects the `slug` variable and returns a JSON payload with the following properties:
- `id`
- `playbackAccessToken.signature`
- `playbackAccessToken.value`
- `videoQualities[].framerate`
- `videoQualities[].quality`
- `videoQualities[].sourceURL`

`playbackAccessToken.signature` needs to be added to the `videoQualities[].sourceURL` as the `sig` query string parameter, and `playbackAccessToken.value` as the `token` query string parameter in order to be able to access clips, which are `mp4` containers.
